cmd/nerdctl/system_prune.go

```diff
@@ -99,5 +102,68 @@ func systemPruneAction(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
-	return imagePrune(cmd, client, ctx)
+	if err := imagePrune(cmd, client, ctx); err != nil {
+		return nil
+	}
+	prunedObjects, err := buildCachePrune(ctx, cmd, all)
+	if err != nil {
+		return nil
+	}
+
+	if len(prunedObjects) > 0 {
+		fmt.Fprintln(cmd.OutOrStdout(), "Deleted build cache objects:")
+		for _, item := range prunedObjects {
+			fmt.Fprintln(cmd.OutOrStdout(), item.ID)
+		}
+	}
+
+	// TODO: print total reclaimed space
+
+	return nil
+}
+
+type cacheUsageInfo struct {
+	ID   string
+	Size int64
+}
+
+func buildCachePrune(ctx context.Context, cmd *cobra.Command, pruneAll bool) ([]cacheUsageInfo, error) {
```

Comment:
Can we have tests?

Reply:
NP! I think it should use `buildkitd` to finish the process, I would find out how to write the integration tests.

Reply:
Hi @ktock , I need help. I did not figure out how to write a test case for the method `buildCachePrune`.

The existing integration test is not exactly the same as what I wondered about, it only makes assertions on the command exit code. 

And there is no other example for other `<something>Prune` methods.

What should I do to write a testcase?

Reply:
This can be done in following-up patches. One of the possible tests will be that running `nerdctl system prune` and checking if the build cache is pruned by running `buildctl`. If it's not easy to allow `go test` accessing to `buildctl`, maybe we can consider to add a subcommand to `nerdctl` for redirecting access to `buildctl` for testing purpose.

Reply:
got that! I would make a try.